tests/test_tree.rs

```diff
@@ -162,4 +159,128 @@ fn recovery() {
     t.__delete_all_files();
 }
 
-// TODO quickcheck splits, reads, writes interleaved
+#[derive(Debug, Clone)]
+enum Op {
+    Set(u8, u8),
+    Get(u8),
+    Del(u8),
+    Cas(u8, u8, u8),
+    Scan(u8, usize),
+    Restart,
+}
+
+impl Arbitrary for Op {
+    fn arbitrary<G: Gen>(g: &mut G) -> Op {
+        if g.gen_weighted_bool(50) {
+            return Op::Restart;
+        }
+
+        let choice = g.gen_range(0, 5);
+
+        match choice {
+            0 => Op::Set(g.gen::<u8>(), g.gen::<u8>()),
+            1 => Op::Get(g.gen::<u8>()),
+            2 => Op::Del(g.gen::<u8>()),
+            3 => Op::Cas(g.gen::<u8>(), g.gen::<u8>(), g.gen::<u8>()),
+            4 => Op::Scan(g.gen::<u8>(), g.gen_range::<usize>(0, 40)),
+            _ => panic!("impossible choice"),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+struct OpVec {
+    ops: Vec<Op>,
+}
+
+impl Arbitrary for OpVec {
+    fn arbitrary<G: Gen>(g: &mut G) -> OpVec {
+        let mut ops = vec![];
+        for _ in 0..g.gen_range(1, 100) {
+            let op = Op::arbitrary(g);
+            ops.push(op);
+
+        }
+        OpVec {
+            ops: ops,
+        }
+    }
+
+    fn shrink(&self) -> Box<Iterator<Item = OpVec>> {
+        let mut smaller = vec![];
+        for i in 0..self.ops.len() {
+            let mut clone = self.clone();
+            clone.ops.remove(i);
+            smaller.push(clone);
+        }
+
+        Box::new(smaller.into_iter())
+    }
+}
```

Comment:
the `shrink` function is used to make a failing test case more manageable to understand. it will generate a shorter sequence of operations, and keep dropping items out until it finds the shortest possible set of operations that trigger the violation of the property